test/e2e/apps/statefulset.go

```diff
@@ -1339,6 +1340,69 @@ var _ = SIGDescribe("StatefulSet", func() {
 			framework.ExpectNoError(err)
 		})
 
+		ginkgo.It("should not delete PVC with OnScaledown policy if another controller owns the PVC", func(ctx context.Context) {
+			e2epv.SkipIfNoDefaultStorageClass(ctx, c)
+			ginkgo.By("Creating statefulset " + ssName + " in namespace " + ns)
+			*(ss.Spec.Replicas) = 3
+			_, err := c.AppsV1().StatefulSets(ns).Create(ctx, ss, metav1.CreateOptions{})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Confirm PVC has been created")
+			err = verifyStatefulSetPVCsExist(ctx, c, ss, []int{0, 1, 2})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Update PVC 1 owner ref")
+			pvc, err := c.CoreV1().PersistentVolumeClaims(ns).Get(ctx, fmt.Sprintf("datadir-%s-1", ss.Name), metav1.GetOptions{})
+			framework.ExpectNoError(err)
+			pvc.OwnerReferences = []metav1.OwnerReference{
+				{
+					Name:       "dummy-controller",
+					APIVersion: "dummy/v1",
+					Kind:       "dummy",
+					UID:        "123",
```

Comment:
can we generate real fake UUID for e2e tests?

Reply:
done